Update asan-known-issues.md (#5340)
MicrosoftDocs/cpp-docs: docs/sanitizers/asan-known-issues.md

## Standard library support

The MSVC standard library (STL) makes partial use of the AddressSanitizer and provides other code safety checks. For more information, see [container-overflow error](./error-container-overflow.md).

> When annotations are disabled, or in versions of the Standard Library that don't support them, AddressSanitizer exceptions raised in STL code still identify real bugs. However, they are more precise if annoations are enabled and you use a version of the Standard Library that supports them.

Reviewer: ```suggestion
When annotations are disabled, or in versions of the Standard Library that don't support them, AddressSanitizer exceptions raised in STL code still identify real bugs. However, they are more precise if annotations are enabled and you use a version of the Standard Library that supports them.
```

Thread local variables (global variables declared with `__declspec(thread)` or `thread_local`) aren't protected by AddressSanitizer. This limitation isn't specific to Windows or Microsoft Visual C++, but is a general limitation.

> ## Loading and unloading DLLs

Reviewer 2: I'm partial to this alternative title.

```suggestion
## Issues with partially sanitized executables.
```

After all, the issue is not with loading and unloading dlls per-se, but instead with the partial `/fsanitize` across components. What do you think?

Reviewer 3: @tylermsft Can you respond to the comment above?

Reviewer 4: This isn't my article. @zacklj89 is writing it. 
My 2c is I like the suggested change.

Author: One reason I didn't use _sanitized_ in the title is there is no previous mention of "sanitized" here (whereas we define ASan, STL, etc). I wanted to avoid confusion by naming the section as the action that is taken when the side effects occur. I'll let @TylerMSFT have the final call.

Reviewer 4: I think the reader can make the leap, given the context of the article, and that the suggested heading more directly addresses why the reader might want to read that section.

> If the entire process is not compiled with `/fsanitize=address`, there may be limitations in ASan's ability to diagnose memory safety errors. The most common example is when a DLL is compiled with ASan and loaded into a process that **was not** compiled with ASan. In this case, ASan will attempt to categorize allocations that took place prior to ASan initialization. If those allocations are transformed in any way (reallocated, moved, etc.), ASan will then attempt to own and monitor the lifetime and memory loads/stores.

Reviewer 2: nit: I think we can be a bit more assertive in the opening sentence:

```suggestion
ASan's analysis is limited when executing alongside binaries that were not complied with `/fsanitize=address`. For example, this would occur if an executable compiled _without_ `/fsanitize=address` loads an ASan-instrumented `.dll`.

In this case, ASan coverage over allocations that took place prior to ASan initialization is best-effort. If those allocations are transformed in any way (reallocated, moved, etc.), ASan will then attempt to own and monitor their lifetime and accesses moving forward.
```

> In the event that all DLLs compiled with ASan leave the process before the end of its execution (like in the event of test drivers such as [TAEF](https://learn.microsoft.com/en-us/windows-hardware/drivers/taef/)), there may dangling references to intercepted functions (`memcmp`, `memcpy`, `memmove`, etc.), and may result in crashes. This is a current limitation with these configurations and unloading DLLs. 

Reviewer 2: in this case, I think the `TAEF` reference may be removed, unless we think we'll get more TAEF reports in the future. I don't feel strongly, but given that it's mostly an internal tool, I'm unsure if there's much value in referencing it externally.

Reviewer 2: > This is a current limitation with these configurations and unloading DLLs.

Is this a limitation of TAEF, or our own. I feel like there's little _we_ could do to improve this, or is there?
